iceoryx_hoofs/test/moduletests/test_cxx_helplets.cpp

```diff
@@ -507,13 +513,163 @@ TEST(Helplets_test_isValidFilePath, WhenOneInvalidCharacterIsContainedPathIsInva
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("a")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("a/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("fuu:blubb")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("fuu:blubb/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/blarbi")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/blarbi/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/x")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/x/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu:-012")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu:-012/")));
+}
+
+TEST(Helplets_test_isValidPath, ValidPathsWithNoRelativeComponentAreValid)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "bf7a0a75-c59e-46a8-96f1-1f848e1c3e43");
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu/bla/blubb/balaa")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/fuu/bla/blubb/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/a/b/c/d/1/2/4")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("/a/b/c/d/1/2/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("asd/fuu/asdaaas/1")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("asd/fuu/asdaaas/")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("123/456")));
+    EXPECT_TRUE(isValidPath(string<FILE_PATH_LENGTH>("123/456/")));
+}
+
+TEST(Helplets_test_isValidPath, EndingWithRelativePathComponentIsInvalid)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "506f9823-39cc-4cbc-b064-84d45b2311e8");
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("./..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("../.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("some/path/to/..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/another/path/to/.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("../bla/fuu/../blubb/.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("./blubb/fuu/../bla/..")));
+}
+
+TEST(Helplets_test_isValidPath, FilePathsWithEndingDotsAreInvalid)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f79660e6-12b5-4ad0-bc26-766da34898b8");
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("a.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/asda.")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/bla/../fuu/asda..")));
+    EXPECT_FALSE(isValidPath(string<FILE_PATH_LENGTH>("/bla/./.././xa..")));
+}
+
+TEST(Helplets_test_isValidPath, PathWhichContainsAllValidCharactersIsValid)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8052b601-c9ad-4cb8-9a87-c301f213d8c4");
+    EXPECT_TRUE(isValidPath(
+        string<FILE_PATH_LENGTH>("/abcdefghijklmnopqrstuvwxyz/ABCDEFGHIJKLMNOPQRSTUVWXYZ/0123456789/-.:_")));
```

Comment:
Does it mean that removing `:_` from the end makes this path invalid?

Reply:
Because the `FilePathsWithEndingDotsAreInvalid` test suggest that this is the case since the path would end with a dot.

Reply:
Ah of course then you are right.

The validation of the file names, paths and entries should work platform independently and therefore I restricted the names a bit to support some more file systems or let's say weird applications.

For instance, NTFS supports file names which end with a dot (`blaa.`) but the windows file explorer does not. So you can see the file but not open, modify or delete it since it is not supported by the Windows API. Yeah.